### DISCUSSION TITLE
how to start code java backend ?

### DISCUSSION THREAD

**Discussion by hoangtien2k3dev** [hoangtien2k3/Java#17] [Q&A] [answered]
I know java code with sql server, now according to backend programming, what should I learn next ?

**Comment by hoangtien2k3** [comment, marked as answer]
If you already know Java and SQL Server, you have a good foundation for backend programming.
1. Spring Framework
2. RESTful Web Services
3. NoSQL Databases
4. Message Queueing
5. Microservices Architecture
6. Containerization
7. Cloud Services
